taropsbt/interface.go

```diff
@@ -139,3 +143,95 @@ func (o *VOutput) SplitLocator(assetID asset.ID) commitment.SplitLocator {
 		Amount:      o.Amount,
 	}
 }
+
+// SetAnchorInternalKey sets the internal key and derivation path of the anchor
+// output based on the given key descriptor and coin type.
+func (o *VOutput) SetAnchorInternalKey(keyDesc keychain.KeyDescriptor,
+	coinType uint32) {
+
+	bip32Derivation, trBip32Derivation := Bip32DerivationFromKeyDesc(
+		keyDesc, coinType,
+	)
+	o.AnchorOutputInternalKey = keyDesc.PubKey
+	o.AnchorOutputBip32Derivation = bip32Derivation
+	o.AnchorOutputTaprootBip32Derivation = trBip32Derivation
+}
+
+// AnchorKeyToDesc attempts to extract the key descriptor of the anchor output
+// from the anchor output BIP32 derivation information.
+func (o *VOutput) AnchorKeyToDesc() (keychain.KeyDescriptor, error) {
+	if o.AnchorOutputBip32Derivation == nil {
+		return keychain.KeyDescriptor{}, fmt.Errorf("anchor output " +
+			"bip32 derivation is missing")
+	}
+
+	return KeyDescFromBip32Derivation(o.AnchorOutputBip32Derivation)
+}
+
+// KeyDescFromBip32Derivation attempts to extract the key descriptor from the
+// given public key and BIP32 derivation information.
+func KeyDescFromBip32Derivation(
+	bip32Derivation *psbt.Bip32Derivation) (keychain.KeyDescriptor, error) {
+
+	if len(bip32Derivation.PubKey) == 0 {
+		return keychain.KeyDescriptor{}, fmt.Errorf("pubkey is missing")
+	}
+
+	pubKey, err := btcec.ParsePubKey(bip32Derivation.PubKey)
+	if err != nil {
+		return keychain.KeyDescriptor{}, fmt.Errorf("error parsing "+
+			"pubkey: %w", err)
+	}
+
+	family, index, err := extractLocatorFromPath(bip32Derivation.Bip32Path)
+	if err != nil {
+		return keychain.KeyDescriptor{}, fmt.Errorf("unable to "+
+			"extract locator from path: %w", err)
+	}
+
+	return keychain.KeyDescriptor{
+		PubKey: pubKey,
+		KeyLocator: keychain.KeyLocator{
+			Family: family,
+			Index:  index,
+		},
+	}, nil
+}
+
+// Bip32DerivationFromKeyDesc returns the default and Taproot BIP32 key
+// derivation information from the given key descriptor information.
+func Bip32DerivationFromKeyDesc(keyDesc keychain.KeyDescriptor,
+	coinType uint32) (*psbt.Bip32Derivation, *psbt.TaprootBip32Derivation) {
+
+	bip32Derivation := &psbt.Bip32Derivation{
+		PubKey: keyDesc.PubKey.SerializeCompressed(),
+		Bip32Path: []uint32{
+			keychain.BIP0043Purpose + hdkeychain.HardenedKeyStart,
+			coinType + hdkeychain.HardenedKeyStart,
+			uint32(keyDesc.Family) +
+				uint32(hdkeychain.HardenedKeyStart),
+			0,
+			keyDesc.Index,
+		},
+	}
+
+	return bip32Derivation, &psbt.TaprootBip32Derivation{
+		XOnlyPubKey:          bip32Derivation.PubKey[1:],
+		MasterKeyFingerprint: bip32Derivation.MasterKeyFingerprint,
+		Bip32Path:            bip32Derivation.Bip32Path,
+	}
+}
+
+// extractLocatorFromPath extracts the key family and index from the given BIP32
+// derivation path.
+func extractLocatorFromPath(path []uint32) (keychain.KeyFamily, uint32, error) {
```

Comment:
Could use a simple comment with the path structure `44/account/index`, etc, etc.

Reply:
I agree, will add.